tests/framework/extensions/rke1/nodepools/nodepools.go

```diff
@@ -57,61 +64,83 @@ func NodePoolSetup(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, Nod
+// AllNodeList is a helper method that will return each node in the cluster.
+func AllNodeList(client *rancher.Client, cluster *management.Cluster, nodeRoles NodeRoles) (*management.Node, error) {
+	nodes, err := client.Management.Node.ListAll(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"clusterId": cluster.ID,
+		},
+	})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if nodestat.AllManagementNodeReady(client, ClusterID) != nil {
-		return err
+	for _, node := range nodes.Data {
+		if nodeRoles.ControlPlane != node.ControlPlane {
+			continue
+		}
+		if nodeRoles.Etcd != node.Etcd {
+			continue
+		}
+		if nodeRoles.Worker != node.Worker {
+			continue
+		}
+
+		return &node, nil
 	}
 
-	logrus.Infof("New node pool is ready!")
-	nodePoolConfig.Quantity = 2
+	return nil, nil
+}
 
-	logrus.Infof("Scaling node pool to 2 worker nodes...")
-	updatedPool, err := client.Management.NodePool.Update(nodePool, &nodePoolConfig)
+// updateNodePoolQuantity is a helper method that will update the node pool with the desired quantity.
+func updateNodePoolQuantity(client *rancher.Client, cluster *management.Cluster, node *management.Node, nodeRoles NodeRoles) (*management.NodePool, error) {
+	updatedNodePool, err := client.Management.NodePool.ByID(node.NodePoolID)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if nodestat.AllManagementNodeReady(client, ClusterID) != nil {
-		return err
+	updatedNodePool.Quantity += nodeRoles.Quantity
+
+	logrus.Infof("Scaling the machine pool to %v total nodes", updatedNodePool.Quantity)
+	_, err = client.Management.NodePool.Update(updatedNodePool, &updatedNodePool)
+	if err != nil {
+		return nil, err
 	}
 
-	logrus.Infof("Node pool is scaled to 2 worker nodes!")
-	nodePoolConfig.Quantity = 1
+	err = kwait.Poll(500*time.Millisecond, 10*time.Minute, func() (done bool, err error) {
+		clusterResp, err := client.Management.Cluster.ByID(cluster.ID)
+		if err != nil {
+			return false, err
+		}
 
-	logrus.Infof("Scaling node pool back to 1 worker node...")
-	_, err = client.Management.NodePool.Update(updatedPool, &nodePoolConfig)
+		if clusterResp.State == active && nodestat.AllManagementNodeReady(client, clusterResp.ID) == nil {
+			logrus.Infof("Node pool is scaled!")
+			return true, nil
+		} else {
+			return false, nil
+		}
+	})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	logrus.Infof("Node pool is scaled back to 1 worker node!")
+	return updatedNodePool, nil
+}
+
+// ScaleNodePoolNodes is a helper method that will add a new node pool to the existing RKE1 cluster, based on the nodeRoles
+// configuration. Once done, it will scale the node pool, scale it back down and then delete the node pool.
+func ScaleNodePoolNodes(client *rancher.Client, cluster *management.Cluster, node *management.Node, nodeRoles NodeRoles) (*management.NodePool, error) {
+	adminClient, err := rancher.NewClient(client.RancherConfig.AdminToken, client.Session)
```

Comment:
Just to make these scale functions more flexible for the future; do we need the admin client or should it use whatever client is passed? It seems like you might wanna add standard users or other users and clients to the test cases later on

Reply:
It can use whatever client, I'll remove this.